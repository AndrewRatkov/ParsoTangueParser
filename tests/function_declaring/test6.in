def void loop(int x)
if x > 0 then
loop(x - 1);
fi
return;
enddef