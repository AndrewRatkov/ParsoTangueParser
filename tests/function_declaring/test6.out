[DEF FUN loop(INTEGER x)=>VOID]
|   
+-->IF
|   |
|   +--->[INT,> ]------->[INT,x]
|   |        |
|   |        +---------->[INT,0]
|   |
|   THEN
|   |
|   +--->[CALL VOID loop]------->[INT,- ]------->[INT,x]
|   |                                |
|   |                                +---------->[INT,1]
|   |
|   FI
|   
+-->[RETURN]
